### PR TITLE
Ria 3305 hearing bundle ready

### PR DESCRIPTION
**Pull request by Lee365**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RIA-3305

### Change description ###
RIA-3305 hearing bundle ready

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/postsubmit/CustomiseHearingBundleConfirmationTest.java [hunk 1]
```diff
@@ -1,16 +1,22 @@
 package uk.gov.hmcts.reform.iacaseapi.domain.handlers.postsubmit;
 
+import static junit.framework.TestCase.assertFalse;
+import static junit.framework.TestCase.assertTrue;
 import static org.assertj.core.api.Assertions.assertThatThrownBy;
 import static org.hamcrest.Matchers.containsString;
-import static org.junit.Assert.*;
+import static org.hibernate.validator.internal.util.Contracts.assertNotNull;
```
Comment (jkumar-moj) [inline src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/postsubmit/CustomiseHearingBundleConfirmationTest.java:7]: Though it works I think we do not need the hibernate library for this. Could you please use assertj library instead to be in consistent.

File: src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/postsubmit/CustomiseHearingBundleConfirmationTest.java [hunk 2]
```diff
@@ -60,6 +69,39 @@ public void should_return_confirmation() {
...
+        when(callback.getEvent()).thenReturn(Event.CUSTOMISE_HEARING_BUNDLE);
+        when(asylumCase.read(AsylumCaseFieldDefinition.HOME_OFFICE_HEARING_BUNDLE_READY_INSTRUCT_STATUS, String.class))
+            .thenReturn(Optional.of("FAIL"));
+
+        PostSubmitCallbackResponse callbackResponse =
+            customiseHearingBundleConfirmation.handle(callback);
+
+        Assert.assertNotNull(callbackResponse);
+        Assertions.assertThat(callbackResponse.getConfirmationHeader()).isNotPresent();
+        Assert.assertTrue(callbackResponse.getConfirmationBody().isPresent());
```
Comment (jkumar-moj) [inline src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/postsubmit/CustomiseHearingBundleConfirmationTest.java:87]: using assertThat from two different libraries. Could we be consistent here? may be use just the assertj library?

File: src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/postsubmit/GenerateHearingBundleConfirmationTest.java [hunk 3]
```diff
@@ -1,16 +1,22 @@
 package uk.gov.hmcts.reform.iacaseapi.domain.handlers.postsubmit;
 
+import static junit.framework.TestCase.assertFalse;
+import static junit.framework.TestCase.assertTrue;
 import static org.assertj.core.api.Assertions.assertThatThrownBy;
 import static org.hamcrest.Matchers.containsString;
-import static org.junit.Assert.*;
+import static org.hibernate.validator.internal.util.Contracts.assertNotNull;
```
Comment (jkumar-moj) [inline src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/postsubmit/GenerateHearingBundleConfirmationTest.java:7]: same as the earlier one. Please remove the dependency on hibernate library

File: src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/postsubmit/GenerateHearingBundleConfirmationTest.java [hunk 4]
```diff
@@ -58,6 +67,39 @@ public void should_return_confirmation() {
...
+        when(callback.getEvent()).thenReturn(Event.GENERATE_HEARING_BUNDLE);
+        when(asylumCase.read(AsylumCaseFieldDefinition.HOME_OFFICE_HEARING_BUNDLE_READY_INSTRUCT_STATUS, String.class))
+            .thenReturn(Optional.of("FAIL"));
+
+        PostSubmitCallbackResponse callbackResponse =
+            generateHearingBundleConfirmation.handle(callback);
+
+        Assert.assertNotNull(callbackResponse);
+        Assertions.assertThat(callbackResponse.getConfirmationHeader()).isNotPresent();
+        Assert.assertTrue(callbackResponse.getConfirmationBody().isPresent());
```
Comment (jkumar-moj) [inline src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/postsubmit/GenerateHearingBundleConfirmationTest.java:85]: Same as earlier one. Use one library for assertThat to be consistent.

**Review by jkumar-moj — CHANGES_REQUESTED** [review]
Hi @Lee365 , few minor comments to fix and also could you please add functional tests for confirmation changes.

File: src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/postsubmit/CustomiseHearingBundleConfirmationTest.java (hunk 1 above)
Comment (Lee365) [reply src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/postsubmit/CustomiseHearingBundleConfirmationTest.java:7]: Corrected Assert statements

File: src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/postsubmit/CustomiseHearingBundleConfirmationTest.java (hunk 2 above)
Comment (Lee365) [reply src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/postsubmit/CustomiseHearingBundleConfirmationTest.java:87]: Corrected Assert statements

File: src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/postsubmit/GenerateHearingBundleConfirmationTest.java (hunk 4 above)
Comment (Lee365) [reply src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/postsubmit/GenerateHearingBundleConfirmationTest.java:85]: Corrected Assert statements

**Review by jkumar-moj — DISMISSED** [review]

**Review by jkumar-moj — DISMISSED** [review]

**Review by Srihari-Nakkala — APPROVED** [review]
LGTM

**Review by bradsorour — APPROVED** [review]
LGTM

**Pull request merged**